src/Documentation~/compatibility.md

```diff
@@ -0,0 +1,17 @@
+# Compatibility with `UnityEngine` mathematics
+
+If you want to Burst-compile your project, it's best practice to use the Unity Mathematics package by default and only use `Mathf` when necessary. Unity Mathematics isn't implemented in the same way as the [`UnityEngine.Mathf`](https://docs.unity3d.com/ScriptReference/Mathf.html) struct in the core Unity engine. Because of this, if your application relies on the specific behaviors of `Mathf`, you'll have to reimplement them to get similar behavior in Unity Mathematics. 
```

Comment:
Feels like this one could just be an instruction - "To Burst-compile your project, use the Mathematics package by default. Only use `Mathf` when necessary" + an example of when it might be necessary to use Mathf instead.

Also, you've said they're implemented differently, but are there things Mathf can do that Mathematics can't, or is Mathf use mainly for legacy compatibility?

Reply:
As far as I understand it, the main purpose of Unity.Mathematics is so that you can use mathematics code with the Burst-compiler and in jobs and such. So if you're using the Entities package for example, you'll want to use Unity.Mathematics to get the most performant maths code possible.

Dale would probably be able to elaborate on whether there are any other advantages/disadvantages of using Mathf though. I suppose in some scenarios it might be easier / friendlier to use the Mathf code if you just want to bash something out and can't be bothered with writing your own random number generator or really complicated maths stuff, but I don't know! 😄

Reply:
Only when you use Unity.Mathematics, do you get the full benefit of Burst vectorizing and optimizing your code. The heavier Mathf functions are written in native C++, and so Burst can't optimize across the function call.